### PR TITLE
test: Add missing unit tests URLEndpointListener

### PR DESCRIPTION
**Pull request by jayahariv**
* add missing unit tests for URLEndpointListener - ObjC & Swift
* fixes: CBL-1477

### CODE REVIEW

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 1]
```diff
@@ -1553,6 +1557,81 @@ - (void) testStopListener {
...
+    AssertEqual(self.otherDB.count, 0);
+    
+    [self runWithTarget: _listener.localEndpoint
+                   type: kCBLReplicatorTypePushAndPull
+             continuous: NO
+          authenticator: [[CBLClientCertificateAuthenticator alloc] initWithIdentity: identity]
+             serverCert: nil
+              errorCode: 0
+            errorDomain: nil];
+
```
Comment (jayahariv) [inline Objective-C/Tests/URLEndpointListenerTest.m:1632]:
```suggestion
```

File: Objective-C/Internal/Replicator/CBLTrustCheck.mm [hunk 2]
```diff
@@ -201,6 +201,7 @@ - (NSURLCredential*) acceptOnlySelfSignedCert: (NSError**)outError {
     if (!isSelfSigned) {
         MYReturnError(outError, NSURLErrorServerCertificateUntrusted, NSURLErrorDomain,
             @"Server SSL certificate is not self-signed");
+        CBLLogVerbose(WebSocket, @"Server SSL certificate is not self-signed, force trusting...");
```
Comment (pasin) [inline Objective-C/Internal/Replicator/CBLTrustCheck.mm:204]: I think there is a bug here that it should return nil instead of doing force trust.

**Review by pasin — CHANGES_REQUESTED** [review]

**Review by pasin — CHANGES_REQUESTED** [review]

**Comment by jayahariv** [comment]
I was using the same name which Android unit test uses(only for those missing tests).

File: Swift/Tests/URLEndpointListenerTest.swift [hunk 3]
```diff
@@ -1058,6 +1060,111 @@ class URLEndpontListenerTest: ReplicatorTest {
...
+            auth: ClientCertificateAuthenticator(identity: identity),
+            acceptSelfSignedOnly: true,
+            serverCert: nil,
+            expectedError: 0)
+        
+        try stopListener(listener: listener!)
+        try TLSIdentity.deleteIdentity(withLabel: serverCertLabel)
+    }
+    
+    func testTLSSelfSignedClientAuthenticatorWithChainedCredentials() throws {
```
[the remark this answers is not in the document]
Comment (jayahariv) [reply Swift/Tests/URLEndpointListenerTest.swift:1233]: this test is chained certs server & accept self signed client, which should reject non-self-signed server.  

`testAcceptSelfSignedWithNonSelfSignedChainedCerts`?

File: Objective-C/Tests/URLEndpointListenerTest.m [hunk 4]
```diff
@@ -1553,6 +1489,212 @@ - (void) testStopListener {
...
+              authenticator: nil
+       acceptSelfSignedOnly: YES
+                 serverCert: nil
+                  errorCode: 0
+                errorDomain: nil];
+    }];
+    
+    [self stopListener: listener];
+}
+
```
Comment (jayahariv) [inline Objective-C/Tests/URLEndpointListenerTest.m:1617]:
```suggestion
```

**Comment by jayahariv** [comment]
Renamed all tests

**Review by pasin — APPROVED** [review]

**Pull request merged**